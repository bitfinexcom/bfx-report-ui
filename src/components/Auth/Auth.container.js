import { connect } from 'react-redux'

import authActions from 'state/auth/actions'
import baseActions from 'state/base/actions'

import Auth from './Auth'

function mapStateToProps(state = {}) {
  return {
    apiKey: state.base.apiKey,
    apiSecret: state.base.apiSecret,
    isShown: state.auth.isShown,
    authStatus: state.auth.authStatus,
  }
}

const mapDispatchToProps = dispatch => ({
  checkAuth: () => dispatch(authActions.checkAuth()),
  setKey: key => dispatch(baseActions.setApiKey(key)),
  setSecret: secret => dispatch(baseActions.setApiSecret(secret)),
})

const AuthContainer = connect(mapStateToProps, mapDispatchToProps)(Auth)

export default AuthContainer
