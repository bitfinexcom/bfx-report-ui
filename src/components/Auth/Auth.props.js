import PropTypes from 'prop-types'
import { intlShape } from 'react-intl'

export const propTypes = {
  authStatus: PropTypes.bool,
  apiKey: PropTypes.string,
  apiSecret: PropTypes.string,
  checkAuth: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
  isShown: PropTypes.bool.isRequired,
  setKey: PropTypes.func.isRequired,
  setSecret: PropTypes.func.isRequired,
}

export const defaultProps = {
  authStatus: null,
  apiKey: '',
  apiSecret: '',
  checkAuth: () => {},
  intl: {},
  isShown: false,
  setKey: () => {},
  setSecret: () => {},
}

export const inputKeyPropTypes = {
  intl: intlShape.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string,
}

export const inputKeyDefaultProps = {
  intl: {},
  label: '',
  onChange: () => {},
  name: '',
  placeholder: '',
  value: '',
}
