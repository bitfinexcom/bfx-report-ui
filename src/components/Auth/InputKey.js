import React, { Fragment } from 'react'
import { injectIntl } from 'react-intl'
import {
  FormGroup,
  InputGroup,
} from '@blueprintjs/core'
import { inputKeyPropTypes, inputKeyDefaultProps } from './Auth.props'

export const InputKey = ({
  intl, label, onChange, name, placeholder, value,
}) => (
  <Fragment>
    <FormGroup
      label={intl.formatMessage({ id: label })}
      labelFor={name}
      labelInfo={intl.formatMessage({ id: 'auth.required' })}
    />
    <InputGroup
      id={name}
      type='text'
      name={name}
      placeholder={intl.formatMessage({ id: placeholder })}
      value={value}
      onChange={onChange}
    />
    <br />
  </Fragment>
)

InputKey.propTypes = inputKeyPropTypes
InputKey.defaultProps = inputKeyDefaultProps

export default injectIntl(InputKey)
