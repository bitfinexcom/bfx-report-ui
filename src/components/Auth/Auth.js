import React, { PureComponent } from 'react'
import { injectIntl } from 'react-intl'
import classNames from 'classnames'
import {
  Button,
  Callout,
  Card,
  Elevation,
  Icon,
  Intent,
} from '@blueprintjs/core'

import { platform } from 'var/config'

import { propTypes, defaultProps } from './Auth.props'
import InputKey from './InputKey'

const classes = classNames(
  'bitfinex-auth',
  'col-xs-12',
  'col-sm-offset-1',
  'col-sm-10',
  'col-md-offset-2',
  'col-md-8',
  'col-lg-offset-3',
  'col-lg-6',
)

class Auth extends PureComponent {
  static propTypes = propTypes

  static defaultProps = defaultProps

  constructor(props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }

  handleClick() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.checkAuth()
  }

  handleChange(event) {
    const { setKey, setSecret } = this.props
    const { name, value } = event.target
    if (name === 'key') {
      setKey(value)
    } else if (name === 'secret') {
      setSecret(value)
    }
  }

  render() {
    const {
      apiKey,
      apiSecret,
      authStatus,
      intl,
      isShown,
    } = this.props
    let showValid = ''
    if (authStatus === true) {
      showValid = (<Icon iconSize={21} icon='tick' intent={Intent.SUCCESS} />)
    } else if (authStatus === false) {
      showValid = (<Icon iconSize={21} icon='cross' intent={Intent.DANGER} />)
    }

    return platform.id === 'local' && isShown ? (
      <div className='row'>
        <Card
          className={classes}
          elevation={Elevation.ZERO}
          interactive
        >
          <h4>
            {intl.formatMessage({ id: 'auth.auth' })}
            &nbsp;
            {showValid}
          </h4>
          <Callout>
            {intl.formatMessage({ id: 'auth.auth.note1' })}
            <a href={platform.KEY_URL} target='_blank' rel='noopener noreferrer'>
              {platform.KEY_URL}
            </a>
            {intl.formatMessage({ id: 'auth.auth.note2' })}
          </Callout>
          <br />
          <InputKey
            label='auth.enterAPIKey'
            name='key'
            placeholder='auth.enterAPIKey'
            value={apiKey}
            onChange={this.handleChange}
          />
          <InputKey
            label='auth.enterAPISecret'
            name='secret'
            placeholder='auth.enterAPISecret'
            value={apiSecret}
            onChange={this.handleChange}
          />
          <p>
            <Button
              name='check'
              intent={Intent.PRIMARY}
              onClick={this.handleClick}
              disabled={!apiKey || !apiSecret}
            >
              {intl.formatMessage({ id: 'auth.checkAuth' })}
            </Button>
          </p>
        </Card>
      </div>
    ) : ''
  }
}

export default injectIntl(Auth)
