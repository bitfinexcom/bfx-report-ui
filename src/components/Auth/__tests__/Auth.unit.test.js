import React from 'react'
import ReactDOM from 'react-dom'
import {
  renderWithIntl,
  rendererCreateWithIntl,
} from 'tests/helper'

import Auth from '../Auth'

test('renders without crashing while hidden', () => {
  const div = document.createElement('div')
  renderWithIntl(<Auth />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('renders without crashing', () => {
  const div = document.createElement('div')
  renderWithIntl(<Auth authStatus isShown />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('render correctly while hidden', () => {
  const component = rendererCreateWithIntl(<Auth />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('render correctly', () => {
  const component = rendererCreateWithIntl(<Auth authStatus isShown />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
