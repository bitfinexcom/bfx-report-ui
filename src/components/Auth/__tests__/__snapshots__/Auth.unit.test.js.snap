// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="row"
>
  <div
    className="bp3-card bp3-interactive bp3-elevation-0 bitfinex-auth col-xs-12 col-sm-offset-1 col-sm-10 col-md-offset-2 col-md-8 col-lg-offset-3 col-lg-6"
  >
    <h4>
      Auth
      Â 
      <svg
        className="bp3-icon bp3-intent-success"
        data-icon="tick"
        height={21}
        style={Object {}}
        viewBox="0 0 20 20"
        width={21}
      >
        <desc>
          tick
        </desc>
        <path
          d="M17 4c-.28 0-.53.11-.71.29L7 13.59 3.71 10.3A.965.965 0 0 0 3 10a1.003 1.003 0 0 0-.71 1.71l4 4c.18.18.43.29.71.29s.53-.11.71-.29l10-10A1.003 1.003 0 0 0 17 4z"
          fillRule="evenodd"
        />
      </svg>
    </h4>
    <div
      className="bp3-callout"
    >
      Visit 
      <a
        href="https://www.bitfinex.com/api"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://www.bitfinex.com/api
      </a>
       to get the readonly API key and secret pair.
    </div>
    <br />
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="key"
      >
        Enter API Key:
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      />
    </div>
    <div
      className="bp3-input-group"
    >
      <input
        className="bp3-input"
        id="key"
        name="key"
        onChange={[Function]}
        placeholder="Enter API Key:"
        style={
          Object {
            "paddingRight": 10,
          }
        }
        type="text"
        value=""
      />
    </div>
    <br />
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="secret"
      >
        Enter API Secret:
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      />
    </div>
    <div
      className="bp3-input-group"
    >
      <input
        className="bp3-input"
        id="secret"
        name="secret"
        onChange={[Function]}
        placeholder="Enter API Secret:"
        style={
          Object {
            "paddingRight": 10,
          }
        }
        type="text"
        value=""
      />
    </div>
    <br />
    <p>
      <button
        className="bp3-button bp3-disabled bp3-intent-primary"
        disabled={true}
        name="check"
        onClick={undefined}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="bp3-button-text"
        >
          Check Auth
        </span>
      </button>
    </p>
  </div>
</div>
`;

exports[`render correctly while hidden 1`] = `""`;
